connect/runtime/src/main/java/org/apache/kafka/connect/runtime/isolation/Plugins.java

```diff
@@ -71,14 +71,44 @@ private static DelegatingClassLoader newDelegatingClassLoader(final List<String>
         return Utils.join(plugins, ", ");
     }
 
-    protected static <T> T newPlugin(Class<T> klass) {
+    protected <T> T newPlugin(Class<T> klass) {
+        ClassLoader classLoader = delegatingLoader.pluginClassLoader(klass.getName());
```

Comment:
By the time we get here, the class should have been generally loaded. Do you think we should simply use the class loader of the class than trying to look up the class loader again?

Reply:
I think it's clearer to pull the classloader from the source of truth directly, even if it costs more 
However, removing the dependency on `delegatingLoader` would make it able to be static again, which is also a good benefit. I think I could go either way on this.

Reply:
The other benefit is the way we are loading converters it could technically be loaded in a different plugin loader that what are keyed in the map since we're loading the ConnectConfig after swapping the class loader in `startTask`. If we prefer to keep it the way it is here, we need to revert that change.